app/src/org/commcare/activities/SyncCapableCommCareActivity.java

```diff
@@ -213,6 +215,24 @@ public void handleFormSendResult(String message, boolean success) {
         }
     }
 
+    public void showRateLimitError() {
+        if (!showRateLimitDialog) {
+            handleFormSendResult(Localization.get("sync.fail.rate.limited.server.error"), false);
+            return;
+        }
+        String title = Localization.get("form.send.rate.limit.error.title");
+        String message = Localization.get("form.send.rate.limit.error.message");
+        StandardAlertDialog dialog = StandardAlertDialog.getBasicAlertDialog(this, title,
+                message, null);
+
+        dialog.setNegativeButton(Localization.get("dialog.do.not.show.again"), (dialog1, which) -> {
+            showRateLimitDialog = false;
```

Comment:
This will mean even if user said he don't wanna see the dialog again, it will appear again in the next session. I think a better behaviour will be to save it in app preferences and then expose a setting that allows user to toggel this behaviour.

Reply:
Makes sense. My initial thought was to make sure user gets a detailed popup in every session.